### DISCUSSION TITLE
希望可以选择按照原有的文件夹体系刮削，不要移动和重命名。

### DISCUSSION THREAD

**Discussion by sbtkggsse** [Yuukiy/JavSP#197] [想法 | Ideas]
开发者您好！
1.目前使用几次发现软件做的很简洁，也很实用，不会太花哨。这很棒！
2.目前发现只能按照既有的方式，扫描/刮削/分类/移动/重命名那些，我个人的话，因为有些文件夹的结构固定，不想去变动。所以非常希望可以选择按照原有的文件夹体系刮削，不要移动或重命名媒体文件。
望采纳~~非常感谢~~

**Comment by sbtkggsse** [comment]
最后四个字前后分别加了 两个“~”，导致划线了，这是个bug吧...

**Reply by Yuukiy** [reply]
这是触发了markdown语法

**Comment by Yuukiy** [comment]
你是PT做种需要保持原有结构么？不移动/重命名影片文件的话，某些情况下Jellyfin等工具可能无法识别封面和nfo

**Reply by fengzhiwen1029** [reply]
可以通过硬链接实现原有结构不动，创建一个新的文件并命名成Jellyfin等工具可以识别的格式来实现。所谓硬链接和指针类似，与原来的文件指向的是同一个物理地址，所以不会增加储存空间的占用。我最近都在手动完成这一步骤来做这一工作，希望大佬能参考一下。

**Reply by mcdullmedia** [reply]
> 可以通过硬链接实现原有结构不动，创建一个新的文件并命名成Jellyfin等工具可以识别的格式来实现。所谓硬链接和指针类似，与原来的文件指向的是同一个物理地址，所以不会增加储存空间的占用。我最近都在手动完成这一步骤来做这一工作，希望大佬能参考一下。

兩個問題.
1. hard link 是指電影文件, 還是其他如劇照, poster nfo?  我認為在 output folder 生成所有meta data, 再 hard link 影片是最優解.
2. 如果是hardlink, 會有一個問題是你不知道這電影是否已經刮削. 其中一辦法是在原folder加一個 flag (例如一個空的txt file), 但並不準確, 又或者很多時share出來的為了不影響PT, 是 read only 的. 這個解決不了. 也可以在目的地加一個hash, 就知道同一個電影是否已經存在.  但這個要更改不少代碼

**Comment by sbtkggsse** [comment]
您好！ 感谢大佬回复！&nbsp; 因为我的文件夹编号是自己编的，比如这样命名：


第一部电影文件夹叫：
0001.天空之城
此文件夹内部的话，放着：
天空之城.1080p.h264等等.mp4&nbsp; &nbsp;
天空之城................................nfo
...............jpg
.....


0002.喜剧之王
.....
..


等等。




比如我现在新增一个电影叫：明天的美好


那么我就在新建一个文件夹“0003.明天的美好”
把明天的美好.mp4放入这个新的文件夹即可，然后用你开发的工具刮削即可，挂出来的东西直接放入“0003.明天的美好”文件夹里面就可以了。


总体来说就是希望有一个选择项，可以选择不要动文件夹结构。或者按你现有软件的设定，自动移动或者重命名文件。


这也许是个不错的功能。


望采纳，谢谢~~





---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月19日(周五) 晚上7:29
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yuukiy/JavSP] 希望可以选择按照原有的文件夹体系刮削，不要移动和重命名。 (Discussion #197)




 
你是PT做种需要保持原有结构么？不移动/重命名影片文件的话，某些情况下Jellyfin等工具可能无法识别封面和nfo

**Comment by Yuukiy** [comment]
没看明白……你是不想要分层的文件夹结构，只想一个文件夹名字点开就是影片？
比方这是目前的效果
```
#整理完成
  ├─君色花音,四ツ葉うらら,関根奈美,音ノ木さくら,七瀬萌,ななせ麻衣
  │  └─[SQTE-196] 引き締まった美体があまりに敏感な美少女のSEX事情
  │          SQTE-196-fanart.jpg
  │          SQTE-196-poster.jpg
  │          SQTE-196.mp4
  │          SQTE-196.nfo
  │
  └─相澤南
      └─[IPX-177] 讓高傲妹妹穿過膝襪露絕對領域癡女玩弄 相澤南
              IPX-177-CD1.mp4
              IPX-177-CD2.mp4
              IPX-177-CD3.mp4
              IPX-177-fanart.jpg
              IPX-177-poster.jpg
              IPX-177.nfo
```
你希望的是这样？
```
引き締まった美体があまりに敏感な美少女のSEX事情
    SQTE-196-fanart.jpg
    SQTE-196-poster.jpg
    SQTE-196.mp4
    SQTE-196.nfo
讓高傲妹妹穿過膝襪露絕對領域癡女玩弄 相澤南
    IPX-177-CD1.mp4
    IPX-177-CD2.mp4
    IPX-177-CD3.mp4
    IPX-177-fanart.jpg
    IPX-177-poster.jpg
    IPX-177.nfo
```

**Comment by sbtkggsse** [comment]
是的，我要下面那种结构。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月19日(周五) 晚上8:15
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yuukiy/JavSP] 希望可以选择按照原有的文件夹体系刮削，不要移动和重命名。 (Discussion #197)




 
没看明白……你是不想要分层的文件夹结构，只想一个文件夹名字点开就是影片？
 比方这是目前的效果
 #整理完成   ├─君色花音,四ツ葉うらら,関根奈美,音ノ木さくら,七瀬萌,ななせ麻衣   │  └─[SQTE-196] 引き締まった美体があまりに敏感な美少女のSEX事情   │          SQTE-196-fanart.jpg   │          SQTE-196-poster.jpg   │          SQTE-196.mp4   │          SQTE-196.nfo   │   └─相澤南       └─[IPX-177] 讓高傲妹妹穿過膝襪露絕對領域癡女玩弄 相澤南               IPX-177-CD1.mp4               IPX-177-CD2.mp4               IPX-177-CD3.mp4               IPX-177-fanart.jpg               IPX-177-poster.jpg               IPX-177.nfo  
你希望的是这样？
 [SQTE-196] 引き締まった美体があまりに敏感な美少女のSEX事情     SQTE-196-fanart.jpg     SQTE-196-poster.jpg     SQTE-196.mp4     SQTE-196.nfo [IPX-177] 讓高傲妹妹穿過膝襪露絕對領域癡女玩弄 相澤南     IPX-177-CD1.mp4     IPX-177-CD2.mp4     IPX-177-CD3.mp4     IPX-177-fanart.jpg     IPX-177-poster.jpg     IPX-177.nfo

**Reply by Yuukiy** [reply]
输出文件夹的格式是可以自己调整的，在配置文件里。例如这是默认的配置，它会生成上面那种格式：
`save_dir = $actress/[$num] $title`
你可以去掉其中表示分层结构的`/`，以及你不想要的女优名字段、番号字段，只保留标题，变成这样
`save_dir = $title`
就会生成你想要的结构了

默认的输出文件夹也是可以改的
`output_folder = #整理完成`
你可以在这里指定一个绝对路径，为你想要的保存影片的地方，比如
`output_folder = D:\影片\已经整理好的`
这样也不会创建`#整理完成`这个文件夹，而是使用你指定的

**Comment by sbtkggsse** [comment]
好的，明白了，感谢您的耐心讲解，谢谢~~



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月19日(周五) 晚上8:27
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Yuukiy/JavSP] 希望可以选择按照原有的文件夹体系刮削，不要移动和重命名。 (Discussion #197)




 
输出文件夹的格式是可以自己调整的，在配置文件里。例如这是默认的配置，它会生成上面那种格式：
 save_dir = $actress/[$num] $title
 你可以去掉其中表示分层结构的/，以及你不想要的女优名字段、番号字段，只保留标题，变成这样
 save_dir = $title
 就会生成你想要的结构了
 
默认的输出文件夹也是可以改的
 output_folder = #整理完成
 你可以在这里指定一个绝对路径，为你想要的保存影片的地方，比如
 output_folder = D:\影片\已经整理好的
 这样也不会创建#整理完成这个文件夹，而是使用你指定的

**Comment by Leelagozero** [comment]
我使用这个，给我将文件结果全部给修改了，文件名字也修改了，而且将视频文件放到了其他地方，因为我的文件名字是有日期的。损失惨重啊！后悔死了使用这个软件。软件的原则是不能改变人家原来的东西，你可以重新建立一个，但是不能改变原来的结果。

**Reply by cpaig** [reply]
好像整理完是有一个文件写了以前的文件名和路径名的，你可以按照那个文件手动改回去，就是有点麻烦了

**Reply by mcdullmedia** [reply]
> 我使用这个，给我将文件结果全部给修改了，文件名字也修改了，而且将视频文件放到了其他地方，因为我的文件名字是有日期的。损失惨重啊！后悔死了使用这个软件。软件的原则是不能改变人家原来的东西，你可以重新建立一个，但是不能改变原来的结果。

you should mount a read only share drive for operation
